pkg/ddl/executor.go

```diff
@@ -1861,6 +1960,18 @@ func (e *executor) AlterTable(ctx context.Context, sctx sessionctx.Context, stmt
 					err = e.AlterTableTTLInfoOrEnable(sctx, ident, ttlInfo, ttlEnable, ttlJobInterval)
 
 					ttlOptionsHandled = true
+				case ast.TableOptionSoftDelete:
+					softDeleteInfoToAlter.Enable = opt.BoolValue
```

Comment:
Is `SoftdeleteInfo.Enable` uncessary. If alter an soft delete table to OFF, we only need to set a table's SoftdeleteInfo to nil